docs/topics/upgrade.md

```diff
@@ -0,0 +1,156 @@
+# Upgrading cluster
+
+Lokomotive cluster runs control plane viz. `apiserver`, `scheduler`, `controller-manager`, `coredns`, `kube-proxy` and `calico` as pods, just like any other workloads. To updrage them it is mostly about changing the image tag, which can be done using `kubectl`.
+
+**NOTE**: This document only covers upgrading from one patch version to another. So if you are on Kubernetes `1.15.1` then this guide explains how to upgrade to `1.15.5`. Since this is patch upgrade it is assumed that there were no major changes and just bug and security fixes.
+
+## Prepare
+
+Be ready with the latest patch versions.
+
+* For `apiserver`, `scheduler`, `controller-manager`, `kube-proxy`
+
+Find your current version:
+
+```bash
+kubectl version --short | grep -i server | cut -d':' -f2-
+```
+
+You can see the latest patch version on the Kubernetes release page: https://github.com/kubernetes/kubernetes/releases. After you find the latest patch version just run following command. For e.g. `export latest_kube="v1.15.5"`.
+
+```
+export latest_kube="<latest patch version of kubernetes>"
+```
+
+* For `coredns` check the latest patch version.
+
+Find your current version:
+
+```
+kubectl -n kube-system get deploy coredns -o jsonpath='{.spec.template.spec.containers[0].image}' | cut -d':' -f2-
+```
+
+You can see the latest patch version on the CoreDNS release page: https://github.com/coredns/coredns/releases. After you find the latest patch version just run following command. For e.g. `export latest_coredns="1.6.5"`.
+
+**Note**: There is no `v` prefixed to the version.
+
+```
+export latest_coredns="<latest patch version of coredns>"
+```
+
+* For `calico` check the latest patch version.
+
+Find your current version:
+
+```
+kubectl -n kube-system get ds calico-node -o jsonpath='{.spec.template.spec.containers[0].image}' | cut -d':' -f2-
+```
+
+You can see the latest patch version on the Calico release page: https://github.com/projectcalico/calico/releases. After you find the latest patch version just run following command. For e.g. `export latest_calico="v3.8.4"`.
+
+```
+export latest_calico="<latest patch version of calico>"
```

Comment:
I think it was clear for the kubernetes version that only patch releases could be updated that way, but I'd like to emphasize in a very clear way that it should be **only** patch releases too.

Reply:
I don't understand what you mean here?

Reply:
I mean, the doc is about upgrading to a kubernetes patch release. But then, it also upgrades calico and coreDNS. And is not clear that those (calico and coreDNS) should be upgraded to patch releases as well. I mean, is not that is not being said, but I'd emphasize that a little bit.

Reply:
Should I mention that in the first paragraph that we are also upgrading coredns and calico?

Reply:
Yes, that seems like a nice idea. Thanks!